documentation/src/main/asciidoc/user_guide/integrations_hibernate_cache.adoc

```diff
@@ -0,0 +1,506 @@
+link:https://docs.jboss.org/hibernate/orm/5.2/userguide/html_single/Hibernate_User_Guide.html#caching-mappings[Hibernate documentation].
+
+Optionally, queries can also be cached but for that query cache needs to be enabled:
+
+.Enable query cache
+[cols="1,10"]
+|===
+| JPA       | `<property name="hibernate.cache.use_query_cache" value="true"/>`
+| Spring    | `spring.jpa.properties.hibernate.cache.use_query_cache=true`     
+|===
+
+NOTE: As well as enabling query cache, forcing a query to be cached requires the query to be made cacheable.
+For example, for JPA queries: `query.setHint("org.hibernate.cacheable", Boolean.TRUE)`.
+
+The best way to find out whether second level cache is working or not is to inspect the statistics.
+By inspecting the statistics you can verify if the cache is being hit, if any new data is stored in cache...etc.
+Statistics are disabled by default, so it is recommended that you enable statistics:
+
+.Enable statistics
+[cols="1,10"]
+|===
+| JPA       | `<property name="hibernate.generate_statistics" value="true" />`
+| Spring    | `spring.jpa.properties.hibernate.generate_statistics=true`      
+|===
+
+
+==== Deployment Scenarios
+
+How to configure Infinispan to be the second level cache provider varies slightly depending on the deployment scenario:
+
+===== Single-Node Standalone Hibernate Application
+
+In standalone library mode, a JPA/Hibernate application runs inside a Java SE application or inside containers that don’t offer Infinispan integration.
+
+Enabling Infinispan second level cache provider inside a JPA/Hibernate application that runs in single node is very straightforward.
+First, make sure the Hibernate Infinispan cache provider is available in the classpath.
+Then, modify the persistence.xml to include these properties:
+
+[source, XML, indent=0]
+----
+<!-- Use Infinispan second level cache provider -->
+<property name="hibernate.cache.region.factory_class" value="infinispan"/>
+
+<!--
+   Force using local configuration when only using a single node.
+   Otherwise a clustered configuration is loaded.
+-->
+<property name="hibernate.cache.infinispan.cfg"
+     value="org/infinispan/hibernate/cache/commons/builder/infinispan-configs-local.xml"/>
+----
+
+By default when running standalone, the Infinispan second-level cache provider uses an Infinispan configuration that’s designed for clustered environments.
+However, Infinispan also provides a configuration designed for local, single node, environments.
+To enable that configuration, set `hibernate.cache.infinispan.cfg` to `org/infinispan/hibernate/cache/commons/builder/infinispan-configs-local.xml` value.
+You can find more about the configuration check the <<default_local_configuration,local configuration>> section.
+
+A simple tutorial showing how to use Infinispan as Hibernate cache provider in a standalone application can be found
+link:https://github.com/infinispan/infinispan-simple-tutorials/tree/master/hibernate-cache/local[here].
```

Comment:
the tutorial should be moved to the new location (assuming it is located here: https://github.com/infinispan/infinispan-simple-tutorials/tree/master/hibernate-cache-local)

Reply:
https://github.com/infinispan/infinispan-simple-tutorials/tree/master/hibernate-cache/local will be the new location once my other PRs have been integrated.